docs/UserGuide.md

```diff
@@ -180,6 +208,85 @@ If your changes to the data file makes its format invalid, AddressBook will disc
 Furthermore, certain edits can cause the AddressBook to behave in unexpected ways (e.g., if a value entered is outside of the acceptable range). Therefore, edit the data file only if you are confident that you can update it correctly.
 </div>
 
+---
+
+### **Log Interaction History**
+
+Records an interaction or communication with a specific contact, such as calls, meetings, or messages.
+
+**Command Format:**  
+`log <ContactID> [d/<Date>] l/<Interaction Details>`
+
+- `d/` Date of the interaction (optional)
+- `l/` Details of the interaction
+
+**Example:**
+- `log 123 d/2024-09-16 l/Discussed property updates`
+
+---
+
+### **Add or Edit Remarks for a Contact**
+
+Adds or updates a remark for a specific contact.
+
+**Command Format:**  
+`remark <index> r/<remark message>`
+
+- `index` is the contact's index in the list.
+- `r/` The message to be added as a remark.
+
+**Example:**
+- `remark 1 r/Interested in waterfront property`
+
+---
+
+### **View Full Contact Details**
+
+Displays detailed information of a specific contact on a single page.
+
+**Command Format:**  
+`view <ContactID>`
+
+- Shows all the contact’s details.
+
+**Example:**
+- `view 123`
+
+![View Contact Example](images/view_contact.png)
+
+---
+
+### **Favorite Contacts**
+
+Marks a contact as a favorite for quick access.
+
+**Command Format:**  
+`favorite <ContactID>`
+
+- Marks the contact with the specified `ContactID` as a favorite.
+
+**Example:**
+- `favorite 123`
+
+---
```

Comment:
favourite [ContactID] when contact id is absent, bring all favourited contacts to the front